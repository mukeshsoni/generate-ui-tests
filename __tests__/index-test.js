import React from "react"
import Enzyme, { shallow, mount } from "enzyme"
import App from "../demo/src/App"
import testGenerator from "../src/index"
import toJson from "enzyme-to-json"

import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() })

test("something", () => {
  // const NewApp = testGenerator(App)
  const props = {
    person: {
      firstName: "Mukesh",
      lastName: "Soni",
      onChange: function onChange() {
        return console.log("person changed")
      }
    }
  }
  const wrapper = mount(<App {...props} />)

  expect(toJson(wrapper)).toMatchSnapshot()
  // wrapper.find("div").simulate("mouseEnter")
  wrapper
    .find('[data-test-id="input_to_test"]')
    .simulate("keyDown", { keyCode: 65, which: 65 })
  wrapper
    .find('[data-test-id="input_to_test"]')
    .simulate("change", { target: { value: "a", checked: false } })
  wrapper
    .find('[data-test-id="input_to_test"]')
    .simulate("keyDown", { keyCode: 13, which: 13 })
  wrapper
    .find('[data-test-id="input_to_test"]')
    .simulate("keyDown", { keyCode: 13, which: 13 })
  wrapper
    .find('[data-test-id="input_to_test"]')
    .at(0)
    .simulate("keyDown", { keyCode: 66, which: 66 })
  // wrapper
  //   .find('[data-test-id="src_todoitem_test_id_2"]')
  //   // .at(0)
  //   .simulate("change", { target: { value: "on", checked: true } })
  expect(toJson(wrapper)).toMatchSnapshot()
})
