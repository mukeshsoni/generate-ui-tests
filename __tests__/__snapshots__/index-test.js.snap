// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`something 1`] = `
<div
  className="App"
  data-test-id="src_app_test_id_0"
  style={
    Object {
      "backgroundColor": "black",
      "color": "#f1f1f1",
    }
  }
>
  <h3
    data-test-id="src_app_test_id_1"
    style={
      Object {
        "fontSize": "3em",
        "fontWeight": 100,
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    todos
  </h3>
  <input
    data-test-id="input_to_test"
    onChange={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "border": "none",
        "fontSize": "1.2em",
        "marginBottom": 20,
        "padding": "0.5em 1em",
        "width": "100%",
      }
    }
    value=""
  />
  <Footer
    filter="all"
    items={Array []}
    onActiveFilterClick={[Function]}
    onAllFilterClick={[Function]}
    onCompletedFilterClick={[Function]}
  />
  <br
    data-test-id="src_app_test_id_2"
  />
  <br
    data-test-id="src_app_test_id_3"
  />
  <br
    data-test-id="src_app_test_id_4"
  />
  <div
    data-test-id="src_app_test_id_5"
    style={
      Object {
        "marginTop": 100,
      }
    }
  >
    <div
      data-test-id="src_app_test_id_6"
      style={
        Object {
          "display": "flex",
          "marginTop": 60,
        }
      }
    >
      <SimpleButton
        data-test-id="btn-bonker"
        onClick={[Function]}
      >
        Crash it ðŸ’”
      </SimpleButton>
    </div>
    <Bonker
      bonk={false}
    />
  </div>
</div>
`;
